os/include/tinyara/partitions.h

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_TINYARA_PARTITIONS_H
+#define __INCLUDE_TINYARA_PARTITIONS_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <stdint.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* The number of kernel partitions */
+#define KERNEL_BIN_COUNT           2
+
+/* The maximum length of path for kernel partition */
+#define KERNEL_PATH_MAX            16
+
+/* The devname format for kernel partition */
+#define KERNEL_DEVNAME_FMT         "/dev/mtdblock%d"
+
+/* The sturcture of partition information */
+struct part_info_s {
+	uint32_t part_size;         /* Partition size */
+	int8_t part_num;            /* Partition number */
+};
+typedef struct part_info_s part_info_t;
```

Comment:
If this type is not used without `struct kernel_data_s`, we don't need to make a structure. We can merge into `struct kernel_data_s`.